src/Book.java

```diff
@@ -1,3 +1,64 @@
+import java.util.Objects;
+
 public class Book {
+    private String nameBook;
+    private Author author;
+    private int year;
+
+    public Book(String nameBook, Author author, int year) {
+        this.nameBook = nameBook;
+        this.author = author;
+        this.year = year;
+    }
+
+    public void setNameBook(String nameBook) {
+        this.nameBook = nameBook;
+    }
+
+    public void setAuthor(Author author) {
+        this.author = author;
+    }
+
+    public void setYear(int year) {
+        this.year = year;
+    }
+
+    public String getNameBook() {
+        return this.nameBook;
+    }
+
+    public Author getAuthor() {
+        return this.author;
+    }
+
+    public int getYear() {
+        return this.year;
+    }
+
+    public String toString() {
+        String bookToStr = this.getNameBook() + " " + author.toString() + " " + Integer.toString(this.getYear());
+        return bookToStr;
+    }
+
+    public boolean equals(Object book1) {
+
+        if (this.getClass() != book1.getClass()) {
```

Comment:
```suggestion
        if (book1 == null || this.getClass() != book1.getClass()) {
```
по контракту метода equals нужно возвращать false, если придёт null. У тебя сейчас будет просто исключение